### PR TITLE
[web] Split initialization of the _flutter namespace and loader object.

### PR DESCRIPTION
**Pull request by ditman**
This makes it possible to more flexibly compose `{{flutter_js}}` and `{{flutter_build_options}}` in @eyebrowsoffire's [bootstrapping improvements PR](https://github.com/flutter/flutter/pull/144434).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by ditman** [comment]
There's several tests covering this gets initialized correctly; not sure if we have a way to test that this is initialized slightly differently though.

**Review by eyebrowsoffire — APPROVED** [review]
LGTM!

**Comment by ditman** [comment]
This is part of our test bootstrap here:

* https://github.com/flutter/engine/blob/eef5247b696eec9cb2c5a0f995ae35cc4f3cd961/lib/web_ui/dev/test_platform.dart#L556-L564

Unfortunately, I cannot add any code that runs *before* the bootstrap; my idea was to

```
_flutter = {
  miscellaneous: 'value',
};
```

then run the rest of the test, but by the time the `main` method of my test is running, `_flutter` and `_flutter.loader` are already defined.

**Comment by ditman** [comment]
Another way is to have `_flutter.miscellaneous = 'value'` defined in the bootstrap before `flutter.js`, and see that nothing breaks, which I'm going to do right now.

**Comment by ditman** [comment]
By tweaking the `test_platform.dart`, actually all tests are testing this feature now. Removing missing test warning.

**Comment by ditman** [comment]
It seems my test to the platform isn't liked by some web tests? Wat?

**Comment by ditman** [comment]
Let me try again reverting 7af7c2c

**Comment by ditman** [comment]
Reproduced locally, and found JS error in strict mode where:

`if (_flutter == null)`

was throwing an `Uncaught ReferenceError: _flutter is not defined` error...

So replaced the initialization to the exact same thing we use in flutter.js:

`if (!window._flutter)`

That should do the trick.

**Comment by ditman** [comment]
Created an issue while investigating this https://github.com/flutter/flutter/issues/144969
